cms/templates/maintenance/_announcement_edit.html

```diff
@@ -0,0 +1,57 @@
+<%page expression_filter="h"/>
+<%inherit file="base.html" />
+<%namespace name='static' file='../static_content.html'/>
+<%!
+from django.utils.translation import ugettext as _
+from openedx.core.djangolib.markup import HTML, Text
+%>
+<%block name="title">${_('Edit Announcement')}</%block>
+<%block name="viewtitle">
+<h3 class="info-course">
+    <span>${_('Edit Announcement')}</span>
+</h3>
+</%block>
+
+<%block name="viewcontent">
+<section class="container maintenance-content">
+    <div class="container studio-view maintenance-form">
+        <form class="form-create" method="post" action="${action_url}">
+            <div class="wrapper-form announcement-container">
+                <input type="hidden" name="csrfmiddlewaretoken" value="${csrf_token}"/>
```

Comment:
These form templates can be greatly simplified if you use the built-in Django form rendering, e.g.,
 
```diff
diff --git a/cms/templates/maintenance/_announcement_edit.html b/cms/templates/maintenance/_announcement_edit.html
index 8915dc4069..4ee4019db9 100644
--- a/cms/templates/maintenance/_announcement_edit.html
+++ b/cms/templates/maintenance/_announcement_edit.html
@@ -19,17 +19,7 @@ from openedx.core.djangolib.markup import HTML, Text
             <div class="wrapper-form announcement-container">
                 <input type="hidden" name="csrfmiddlewaretoken" value="${csrf_token}"/>
                 <div class="announcement-item">
-                    <textarea name="content" class="announcement-content">
-                        ${getattr(announcement, "content", "") | n}
-                    </textarea>
-                    <div class="actions-list">
-                        <label for="active">${_('Is Active')}</label>
-                        <input type="checkbox" name="active" class="announcement-active"
-                            %if announcement.active:
-                                checked
-                            %endif
-                         />
-                    </div>
+                ${form.as_ul() | n}
                 </div>
                 <div class="actions">
                     <button type="submit" class="action action-primary">${_('Save')}</button>
```

Reply:
+1